nbconvert/preprocessors/execute.py

```diff
@@ -268,6 +282,8 @@ def start_new_kernel(self, **kwargs):
                 'kernelspec', {}).get('name', 'python')
         km = self.kernel_manager_class(kernel_name=self.kernel_name,
                                        config=self.config)
+        if km.ipykernel and self.ipython_hist_file:
```

Comment:
Is `ipykernel` an attribute of a KernelManager? It doesn't seem to be https://jupyter-client.readthedocs.io/en/stable/api/manager.html#jupyter_client.KernelManager

If not, I'd prefer to know where this is being assigned.

Reply:
It [is a property of the Kernel Manager](https://github.com/jupyter/jupyter_client/blob/master/jupyter_client/manager.py#L93) but it isn't documented. I used it to encapsulate the logic about using an `ipython` specific argument, rather than doing my own check against the kernel name before passing this argument in.